### PR TITLE
Actix web 2.0

### PR DESCRIPTION
**Pull request by LucasPickering**
Upgraded actix to 2.0 (async!). Also fixed some websocket stuff. Turns out that multiple route params get passed in as a tuple instead of separate params. The changes were pretty simple, just had to add async to some stuff. I also moved the server launch code into `server/mod.rs`. The only thing that took some work to upgrade was the websockets, but it's still mostly the same.

### CODE REVIEW

File: api/src/server/websocket.rs [hunk 1]
```diff
@@ -174,25 +174,31 @@ impl Actor for ProgramWebsocket {
                 ctx.stop();
             } else {
                 // Not timed out, send another ping
-                ctx.ping("");
+                ctx.ping(b"");
```
Comment (JRMurr) [inline api/src/server/websocket.rs]: this b string makes me mad for no reason lmao

File: api/src/server/websocket.rs [hunk 2]
```diff
@@ -201,34 +207,35 @@ impl StreamHandler<ws::Message, ws::ProtocolError> for ProgramWebsocket {
...
+            Ok(ws::Message::Close(_)) => {
                 ctx.stop();
             }
-            ws::Message::Nop => {}
+
+            // Don't do anything with these messages
+            Ok(ws::Message::Binary(_))
+            | Ok(ws::Message::Continuation(_))
+            | Ok(ws::Message::Nop)
+            | Err(_) => {}
```
Comment (JRMurr) [inline api/src/server/websocket.rs]: Could we unwrap msg before the match so each arm does not need the ok?

File: api/src/server/websocket.rs (hunk 1 above)
Comment (LucasPickering) [reply api/src/server/websocket.rs]: triggered by python?

File: api/src/server/websocket.rs (hunk 2 above)
Comment (LucasPickering) [reply api/src/server/websocket.rs]: Actually, I think I prefer having it this way. Otherwise, you need a nested `match`, which IMO is harder to read.

**Pull request merged**